API.md

```diff
@@ -0,0 +1,337 @@
+hence it does not take the contents of `OutPoint` into account to see if two instances are the same.
+By default, two instances are equals if they point to the same object,
+not if the contents are the same.
+For anything you want as key in a map or dictionary, 
+you want the latter property.
+Whenever you override `equals`, 
+also override `hashCode` to avoid surprises in structures such as hash maps.
+Unless you're using a Java version that includes records,
+I normally let the IDE (e.g. IntelliJ) generate them for me.
+They allow you to select which fields to use for computing `equals`/`hashCode`.
+More details [here](https://www.baeldung.com/java-equals-hashcode-contracts).
+
+
+## Finalizers
+
+I've noticed that some classes, e.g. `OutPut` rely on Java finalizers.
+Finalizers in Java are generally considered a bad idea.
+[This article](https://www.baeldung.com/java-finalize) explains the pitfalls in detail.
+
+The best practice for resource management in Java these days,
+is to make your classes extend `AutoCloseable`, 
+which then enables you to pass them into a try-with-resources construct.
+See [detailed explanation](https://www.baeldung.com/java-try-with-resources).
+
+
+## NioPeerHandler
+
+Assuming the current status quo (using Java NIO directly),
+how does someone close the sockets that `NioPeerHandler` uses?
+As per the previous section, 
+the general consensus is that it should extend `AutoCloseable`,
+and close any socket or other resources it has created in the `close` method implementation.
+
+
+## Style
+
+The JDK and majority of Java libraries use camel case style throughout their source.
+That means it often applies both to method names, variable names, class names...etc.
+Snake case is seldom used, though I've seen it used very sparingly in Java source code that is not exposed to users.
+
+
+## `TwoTuple` vs Nominal Tuples
+
+With Java records, 
+the Java community has a made a decision towards avoiding box standard tuple types.
+
+Even though LDK is unlikely to be based on a Java version that has records in it,
+I would stay away from adding tuple types.
+Instead, I would just create nominal classes.
+
+So, I would remove `TwoTuple`.
+
+
+## Documentation
+
+ChannelManager fresh node example outdated 
+[here](https://lightningdevkit.org/docs/build_node#initialize-the-channelmanager).
+`LDKNetwork` is no longer passed as first parameter, but as 3rd parameter before last.
+
+Code in `channel_manager_constructor.chain_sync_completed();`
+looks to be related to the 
+["Sync ChannelMonitors and ChannelManager to chain tip" section](https://lightningdevkit.org/docs/build_node/#sync-channelmonitors-and-channelmanager-to-chain-tip).
+However, it currently appears under ["Give ChannelMonitors to ChainMonitor#" section](https://lightningdevkit.org/docs/build_node/#give-channelmonitors-to-chainmonitor),
+which does not have any code in it.
+
+
+## params_latest_hash_arg
+
+When constructing a new `ChannelManager`, 
```

Comment:
Hmm, right, the auto-conversion here is just really awkward. Maybe the simplest is to just document that people should use `ChannelManagerConstructor`?